docs(troubleshoot): Add docker error troubleshooting guide (#4972)
flyteorg/flyte: docs/community/troubleshoot.rst

Debugging common execution errors
----------------------------------

``Error: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> This error will show if your are not running Docker with the native Docker engine in a Linux machine. Most probably you are running Docker via Docker Desktop.

Reviewer: ```suggestion
This error will show if you are not running Docker with the native Docker engine in a Linux machine. Most likely you are running Docker via Docker Desktop.
```

> - If you are using Docker Desktop in MacOs, run

Reviewer: ```suggestion
- If you are using Docker Desktop in MacOS, run:
```

.. prompt:: bash $

 sudo ln -s ~/Library/Containers/com.docker.docker/Data/docker.raw.sock /var/run/docker.sock

> - If you are using Docker Desktop in Linux, run

Reviewer: ```suggestion
- If you are using Docker Desktop in Linux, run:
```

.. prompt:: bash $

 sudo ln -s ~$USER/.docker/desktop/docker.sock /var/run/docker.sock

> - If you are using some other tools to run Docker, you need to make sure that ``/var/run/docker.sock`` is linked to the correct socket file.

Reviewer: ```suggestion
- If you are using another tool to run Docker, make sure that ``/var/run/docker.sock`` is linked to the correct socket file.
```

>   For example, if you are using Rancher Desktop on Linux, run

Reviewer: ```suggestion
  For example, if you are using Rancher Desktop on Linux, run:
```
